.github/workflows/build-app.yml

```diff
@@ -0,0 +1,35 @@
+name: Build App
+
+on:
+    workflow_call:
+        inputs:
+            ref:
+                type: string
```

Comment:
Is there a reason to not make this required?

Reply:
For the [build workflow](https://github.com/NordicSemiconductor/pc-nrfconnect-boilerplate/blob/add_artifactory_github_actions/.github/workflows/build.yml) the default is always correct. For the [release workflow](https://github.com/NordicSemiconductor/pc-nrfconnect-shared/blob/artifactory/github_actions/.github/workflows/release-app.yml) the default will be correct most of the time. So we only need to override occasionally.